### PR TITLE
Fix for empty deployUrl check in deployment_create

### LINKED ISSUE #2645: `waypoint deploy`'s logs reference horizon features even with the url service disabled
**Steps to Reproduce**

- Run a waypoint server with `-url-enabled=false`
- Deploy an app
- Observe the following logs:

```
$ waypoint deploy

» Deploying backend...
✓ Kubernetes client connected to <redacted> with namespace dev
✓ Creating deployment...
✓ Expected "http" port "8080" for app "backend-v66"
✓ Deployment successfully rolled out!

✓ Finished building report for Kubernetes platform
✓ Finished building report for Kubernetes deployment resource
✓ Deployment "backend-v66" is reporting ready!

» Releasing...
✓ Kubernetes client connected to <redacted> with namespace dev
✓ Updating existing service...
✓ Service is ready!

» Pruning old deployments...
  Deployment: 01FKETA24EYCMW0TACPC0A236N (v64)
✓ Kubernetes client connected to <redacted> with namespace dev
✓ Deployment deleted

✓ Finished building report for Kubernetes platform
✓ Finished building report for Kubernetes service resource

The deploy was successful! A Waypoint deployment URL is shown below. This
can be used internally to check your deployment and is not meant for external
traffic. You can manage this hostname using "waypoint hostname."

   Release URL: http://internal-a9cnotarealurl6b8224960304da4a-808601300.us-east-2.elb.amazonaws.com
Deployment URL: https://

```

Note that `You can manage this hostname using "waypoint hostname."` is false here, and the Deployment URL is blank

**Expected behavior**

We should not include a deployment URL or reference the `hostname` command if the url service is disabled

**Waypoint Platform Versions**
Additional version and platform information to help triage the issue if
applicable:

* Waypoint CLI Version: 0.6.1
* Waypoint Server Platform and Version: 0.6.1

### PR DESCRIPTION
**Pull request by gunadhya**
Fixes #2645

I'm new to this project so please let me know if any updates are needed.

### CODE REVIEW

File: internal/cli/deployment_create.go [hunk 1]
```diff
@@ -128,10 +128,14 @@ func (c *DeploymentCreateCommand) Run(args []string) int {
 		app.UI.Output("")
 		switch {
 		case releaseUrl != "":
-			printInplaceInfo(inplace, app)
-			app.UI.Output("   Release URL: %s", releaseUrl, terminal.WithSuccessStyle())
-			app.UI.Output("Deployment URL: https://%s", deployUrl, terminal.WithSuccessStyle())
-
+			if deployUrl != "" {
```
Comment (xiaolin-ninja) [inline internal/cli/deployment_create.go:131]: Thanks for making the PR @gunadhya and you're on the right track! We already have a [template](https://github.com/hashicorp/waypoint/blob/95ec521fe35a8b57df49d9e6e33404a57ee17a83/internal/cli/deployment_create.go#L210) defined when URL service is disabled, I would suggest refactoring the case switch statement to hit that code.

**Review by xiaolin-ninja — CHANGES_REQUESTED** [review]

File: internal/cli/deployment_create.go (hunk 1 above)
Comment (gunadhya) [reply internal/cli/deployment_create.go:131]: Oh! Do you mean a new `case` like this?
```go
    case releaseUrl != "" && deployUrl == "":
	app.UI.Output(strings.TrimSpace(deployNoURL)+"\n", terminal.WithSuccessStyle())
	app.UI.Output("   Release URL: %s", releaseUrl, terminal.WithSuccessStyle())
```

Comment (xiaolin-ninja) [reply internal/cli/deployment_create.go:131]: I meant a `case deployUrl == ""` that wouldn't print out the release URL, but thinking more about it we do want the release URL too, so an if statement inside the `releaseUrl` case was the proper approach, I'd do something like

```
case releaseUrl != "":
    printInplaceInfo(inplace,app)
    app.UI.Output(" Release URL: %s", releaseUrl, terminal.WithSuccessStyle())
    if deployUrl != "" {
        app.UI.Output("Deployment URL: https://%s", deployUrl 
    } else {
        app.UI.Output(strings.TrimSpace(deployNoURL)+"\n", terminal.WithSuccessStyle())
    }
```

**Review by xiaolin-ninja — APPROVED** [review]
LGTM!

**Review by briancain — COMMENTED** [review]
Hey there @gunadhya - could you please add a changelog? Our guide on how to do this is here: https://github.com/hashicorp/waypoint/blob/main/.github/CHANGELOG_GUIDE.md

This is a `bug` fix type

**Comment by gunadhya** [comment]
I've added a changelog. If there are any other beginner/Intermediate issues in this project, I'd be happy to work on them.

File: .changelog/2645.txt [hunk 2]
```diff
@@ -0,0 +1,3 @@
+```release-note:bug
+internal/cli/deployment_create: Added check for empty deployUrl in output for release switch case
```
Comment (briancain) [inline .changelog/2645.txt:2]:
```suggestion
cli: Added check for empty deployUrl in output for release switch case 
```

Comment (briancain) [reply .changelog/2645.txt:2]: We usually keep these pretty simple! I'll go ahead and merge this suggestion.

**Review by briancain — APPROVED** [review]

**Review by briancain — COMMENTED** [review]
Hey @gunadhya - The changelog file name should match this pull request, so it should be updated to

> .changelog/2755.txt

Thanks!

Otherwise I think we're good to merge.

**Comment by gunadhya** [comment]
Oh I'll get right on it. Does the documentation needed to be updated https://github.com/hashicorp/waypoint/blob/main/.github/CHANGELOG_GUIDE.md#changelog-entry-examples 
Cause over there it suggested issue number.

**Comment by briancain** [comment]
Oops, yep, that should be updated! Sorry about that ❤️

File: .changelog/2755.txt [hunk 3]
```diff
@@ -0,0 +1,4 @@
+```release-note:bug
+cli: Added check for empty deployUrl in output for release switch case
+CHANGELOG_GUIDE: Updated issue number to pull number
```
Comment (briancain) [inline .changelog/2755.txt:3]:
```suggestion
```

We don't include changelog guide updates in our changelog ❤️ I'll get this comitted, and then merge the PR!

**Comment by briancain** [comment]
> I've added a changelog. If there are any other beginner/Intermediate issues in this project, I'd be happy to work on them.

Hmm, will have to get back to you on that one! If you see something interesting in our backlog of issues, please feel free to drop a comment on it to discuss and we can figure something out before you get started on an issue 😄

**Pull request merged**